src/libraries/System.Linq/src/System/Linq/Max.cs

```diff
@@ -482,12 +484,111 @@ public static decimal Max(this IEnumerable<decimal> source)
                     }
 
                     value = e.Current;
+                    if (comparer == Comparer<TSource>.Default)
```

Comment:
@EgorBo, is this still going to be needed after your change?

Reply:
For this case - yes, unfortunately. We don't know result of `comparer ??= Comparer<TSource>.Default;` at the JIT time so the variable can be anything.

Reply:
What change, out of curiosity?

Reply:
https://github.com/dotnet/runtime/pull/50446